### ISSUE TITLE
(aws-cdk/aws-appsync): GraphqlApi fails to deploy initially due to inability to assume service linked role

### ISSUE THREAD

**Issue by sensedata1** [aws/aws-cdk#16598]
First time deployment of an `appsync.GraphqlApi` with prop: `x-ray: true` results in failure to deploy as `appsync.amazonaws.com` attempts `eventName:CreateGraphqlApi` with message:

```sh
"message":"Failed to assume service-linked role arn:aws:iam::XXXXXXXXXXXX:role/aws-service-role/appsync.amazonaws.com/AWSServiceRoleForAppSync, please retry"
```

When `x-ray` is set to `true`, a service linked role is created behind the scenes, and it seems that this role is either not in place, or does not have the correct permissions in place before appsync tries to use it to createGraphqlApi.

Once an attempt has been made once, the newly created service linked role persists (even if the stack is destroyed), meaning that subsequent attempts to deploy the appsync stack will succeed. 

Attempting to create the service linked role using CDK is not successful. 

### Reproduction Steps

The `GraphqlApi` is deployed as part of an API stack which includes an APIgateway. 
Below is the configuration of the `appsync.GraphqlApi` we're using.

```ts
    const appSync = new appsync.GraphqlApi(this, 'AppSync', {
      authorizationConfig: {
        additionalAuthorizationModes: [
          {
            apiKeyConfig: {
              expires: cdk.Expiration.after(cdk.Duration.days(365)),
            },
            authorizationType: appsync.AuthorizationType.API_KEY,
          },
        ],
        defaultAuthorization: {
          authorizationType: appsync.AuthorizationType.USER_POOL,
          userPoolConfig: {
            userPool,
          },
        },
      },
      name: 'XXX-XXXXX',
      schema: appsync.Schema.fromAsset(schemaPath),
      xrayEnabled: true,
    });
```

### What did you expect to happen?

The `GraphqlApi` should be created successfully

### What actually happened?

```sh
Failed to assume service-linked role arn:aws:iam::XXXXXXXXXXXX:role/aws-service-role/appsync.amazonaws.com/AWSServiceRoleForAppSync, please retry. (Service: AWSAppSync; Status Code: 403; Error Code: AccessDeniedException; Request ID: 30c42675-ab80-4a2b-9ae5-51ca30bfaa9e; Proxy: null)
```

### Environment

  - **CDK CLI Version  : 1.123.0 (build fad855e)**
  - **Framework Version:**
  - **Node.js Version: v14.17.6** 
  - **OS               : macOS Mojave**
  - **Language (Version):** 

### Other

--- 

This is :bug: Bug Report

**Comment by otaviomacedo** [comment]
Hi, @sensedata1. Thanks for raising this. From your description, it does look like an ordering issue.

I am unassigning and marking this issue as `p2`, which means that we are unable to work on this immediately.

We use +1s to help prioritize our work, and are happy to revaluate this issue based on community feedback. You can reach out to the cdk.dev community on [Slack](https://join.slack.com/t/cdk-dev/shared_invite/zt-mso6p56d-qJp7SOTBvMaQuDrx7R2wHg) to solicit support for reprioritization.

**Comment by corymhall** [comment]
This seems like it might be a bug in the CloudFormation resource. If a service role is required in order to create the resource then that should be part of the CloudFormation resource creation. I've created in issue in the [CloudFormation roadmap](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/1315)

I would be hesitant to try and provide a solution in the CDK since working with service linked roles can be tricky. For example, if we added a `CfnServiceLinkedRole` to the appsync library it may also lead to failed deployments if the service linked role already exists in the account with different attributes.

**Comment by jumic** [comment]
Has anyone else received this error in the last few months?

@corymhall Your issue has been resolved in the CloudFormation roadmap. I just removed the service linked role `AWSServiceRoleForAppSync` from my account. I then deployed a new AppSync API with xray enabled. The API was deployed directly and the `AWSServiceRoleForAppSync` role was recreated. So I can't reproduce the issue anymore. Can we close this issue in this case?
